# Type: feat
# Subtype: Ex
# English Summary: Add language mode
# Chinese Summary: 添加语言模式
# Entry Number: 16

# === Git Diff Content ===

diff --git a/src/com/google/javascript/jscomp/Compiler.java b/src/com/google/javascript/jscomp/Compiler.java
index 50a6428..485fcc2 100644
--- a/src/com/google/javascript/jscomp/Compiler.java
+++ b/src/com/google/javascript/jscomp/Compiler.java
@@ -2619,6 +2619,8 @@ public class Compiler extends AbstractCompiler implements ErrorHandler {
   }
 
   private void processNewScript(JsAst ast, Node originalRoot) {
+    languageMode = options.getLanguageIn();
+
     Node js = ast.getAstRoot(this);
     Preconditions.checkNotNull(js);
 
